### PR TITLE
Split Formation assignment error and configuration

### PR DESCRIPTION
**Pull request by StanislavStefanov**
**Description**
Keeping the configuration and errors in the same column for formation assignments is errorprone. Also we can not keep old configurations in case of error(the error overrides the configuration).

Changes proposed in this pull request:
- separate configuration and error in different columns
- expose two new properties for the formation assignments - configuration and error
- calculate the value of the value property of formation assignments - If error != nil -> take that with priority, otherwise return the configuration
- enrich e2e tests to validate the new properties 

**Pull Request status**

- [x] Implementation
- [x] Unit tests
- [x] Integration tests
- [x] `chart/compass/values.yaml` is updated 
- [ ] Mocks are regenerated, using the automated script

### CODE REVIEW

File: tests/director/tests/formation_api_test.go [hunk 1]
```diff
@@ -5491,10 +5516,12 @@ func assertFormationAssignmentsAsynchronously(t *testing.T, ctx context.Context,
 
 		assignmentExpectation, ok := targetAssignmentsExpectations[assignment.Target]
 		require.Truef(t, ok, "Could not find expectations for assignment with ID: %q, source %q and target %q", assignment.ID, assignment.Source, assignment.Target)
-
+		fmt.Printf("assignment with ID: %q, source %q and target %q", assignment.ID, assignment.Source, assignment.Target)
```
Comment (ivantenevvasilev) [inline tests/director/tests/formation_api_test.go:5519]: Maybe change to `t.Logf`?

Comment (StanislavStefanov) [reply tests/director/tests/formation_api_test.go:5519]: It was a debugging message

**Review by ivantenevvasilev — DISMISSED** [review]

**Review by ivantenevvasilev — APPROVED** [review]

**Pull request merged**